bears/coffee_script/CoffeeLintBear.py

```diff
@@ -4,6 +4,11 @@
 from coalib.bears.requirements.NpmRequirement import NpmRequirement
 from coalib.results.RESULT_SEVERITY import RESULT_SEVERITY
 from coalib.results.Result import Result
+from coalib.bearlib.spacing.SpacingHelper import SpacingHelper
+
+
+def _coffee_bool(py_bool):
+    return str(py_bool).lower()
```

Comment:
Oh wow! I guess the reason is same as why creating list using ` [] ` is
faster than `list ()` ? Nice!
On 24 Jun 2016 21:07, "Mischa Krüger" <notifications@github.com> wrote:

> In bears/coffee_script/CoffeeLintBear.py
> <https://github.com/coala-analyzer/coala-bears/pull/555#discussion_r68416895>
> :
>
> > @@ -4,6 +4,11 @@
> >  from coalib.bears.requirements.NpmRequirement import NpmRequirement
> >  from coalib.results.RESULT_SEVERITY import RESULT_SEVERITY
> >  from coalib.results.Result import Result
> > +from coalib.bearlib.spacing.SpacingHelper import SpacingHelper
> > +
> > +
> > +def _coffee_bool(py_bool):
> > +    return str(py_bool).lower()
>
> This is faster :)
>
> def _coffee_bool(py_bool):
>     return "true" if py_bool else "false"
>
> proof:
>
> >>> def coffee_bool(py_bool):
> ...  return str(py_bool).lower()
> ...
> >>> def fastbool(py_bool):
> ...  return "true" if py_bool else "false"
> ...
> >>> from timeit import Timer
> >>> def cb():
> ...  [coffee_bool(x % 2 == 0) for x in range(1000)]
> ...
> >>> cb()
> >>> def fb():
> ...  [fastbool(x % 2 == 0) for x in range(1000)]
> ...
> >>> CB = Timer(cb)
> >>> FB = Timer(fb)
> >>> count = 10000
> >>> CB.timeit(count)
> 5.452193706195506
> >>> FB.timeit(count)
> 2.382442419382003
> >>> CB.timeit(count)
> 5.2724452325734905
> >>> FB.timeit(count)
> 2.532276357108298
>
> so about 50% faster 👍 :D
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/coala-analyzer/coala-bears/pull/555/files/737311530c49c93a4e2b657e0243876317b35005#r68416895>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AHDgOc7ap2hV4iLS4gPS7wkh0-nSx2Y3ks5qO_nTgaJpZM4I9f6t>
> .
>